core/vm/gas_table.go

```diff
@@ -404,82 +404,94 @@ func gasExpEIP158(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memor
 
 func gasCall(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (*multigas.MultiGas, uint64, error) {
 	var (
-		gas            uint64
+		multiGas       = multigas.ZeroGas()
 		transfersValue = !stack.Back(2).IsZero()
 		address        = common.Address(stack.Back(1).Bytes20())
 	)
+
 	if evm.chainRules.IsEIP158 {
 		if transfersValue && evm.StateDB.Empty(address) {
-			gas += params.CallNewAccountGas
+			multiGas.SafeIncrement(multigas.ResourceKindComputation, params.CallNewAccountGas)
 		}
 	} else if !evm.StateDB.Exist(address) {
-		gas += params.CallNewAccountGas
+		multiGas.SafeIncrement(multigas.ResourceKindComputation, params.CallNewAccountGas)
 	}
+
 	if transfersValue && !evm.chainRules.IsEIP4762 {
-		gas += params.CallValueTransferGas
+		multiGas.SafeIncrement(multigas.ResourceKindComputation, params.CallValueTransferGas)
 	}
-	multiGas, memoryGas, err := memoryGasCost(mem, memorySize)
+
+	memoryMultiGas, _, err := memoryGasCost(mem, memorySize)
 	if err != nil {
 		return multigas.ZeroGas(), 0, err
 	}
-	// TODO(NIT-3484): Update multi dimensional gas here
-	var overflow bool
-	if gas, overflow = math.SafeAdd(gas, memoryGas); overflow {
+	multiGas, overflow := multiGas.SafeAdd(multiGas, memoryMultiGas)
+	if overflow {
 		return multigas.ZeroGas(), 0, ErrGasUintOverflow
 	}
+
 	if evm.chainRules.IsEIP4762 && !contract.IsSystemCall {
 		if transfersValue {
-			gas, overflow = math.SafeAdd(gas, evm.AccessEvents.ValueTransferGas(contract.Address(), address))
-			if overflow {
+			valueTransferGas := evm.AccessEvents.ValueTransferGas(contract.Address(), address)
+			if overflow := multiGas.SafeIncrement(multigas.ResourceKindStorageAccess, valueTransferGas); overflow {
 				return multigas.ZeroGas(), 0, ErrGasUintOverflow
 			}
 		}
 	}
-	evm.callGasTemp, err = callGas(evm.chainRules.IsEIP150, contract.Gas, gas, stack.Back(0))
+
+	// NOTE(NIT-3483): using only the computation gas for callGas
```

Comment:
Needs to be double-checked

Reply:
I think the third parameter to `callGas` below (called `base` in that function) should be the sum of all dimensions, not just the computation gas, and with my earlier comment on how `CallNewAccountGas` needs to be storage growth, that will actually matter

Reply:
Thank you, now I figured out. Initially I thought that `memoryGasCost` only makes sense for computation gas because I was getting inconsistent results in tests, which was due to the wrong type for `CallNewAccountGas`. Fixed both issues with latest commit